pkg/unikontainers/hypervisors/firecracker.go

```diff
@@ -62,11 +62,18 @@ type FirecrackerNet struct {
 	HostIF   string `json:"host_dev_name"`
 }
 
+type FirecrackerVSockDev struct {
+	GuestCID int    `json:"guest_cid"`
+	UDSPath  string `json:"uds_path"`
+	VSockID  string `json:"vsock_id"`
+}
+
 type FirecrackerConfig struct {
 	Source  FirecrackerBootSource `json:"boot-source"`
 	Machine FirecrackerMachine    `json:"machine-config"`
 	Drives  []FirecrackerDrive    `json:"drives"`
 	NetIfs  []FirecrackerNet      `json:"network-interfaces"`
+	VSock   FirecrackerVSockDev   `json:"vsock"`
```

Comment:
We need to add the `omitempty` in the json, because otherwise there will always be an entry for a vsocket even if not defined.